src/node/desktop/src/main/detect-r.ts

```diff
@@ -212,6 +215,31 @@ export function detectREnvironment(rPath?: string): Expected<REnvironment> {
   Sys.getenv("${kLdLibraryPathVariable}")
 ))`;
 
+  logger().logDebug(`Querying information about R executable at path: ${rExecutable}`);
+
+  // remove R-related environment variables before invoking R
+  const envCopy = Object.assign({}, process.env);
+  delete envCopy['R_HOME'];
+  delete envCopy['R_ARCH'];
+  delete envCopy['R_DOC_DIR'];
+  delete envCopy['R_INCLUDE_DIR'];
+  delete envCopy['R_RUNTIME'];
+  delete envCopy['R_SHARE_DIR'];
+
+  // On Windows, unset temporary directory environment variables: the R docs state that 
+  // when using R from command-line on Windows:
+  //
+  //   https://cran.r-project.org/doc/manuals/R-intro.html
+  //
+  //   "You need to ensure that either the environment variables TMPDIR, TMP, and TEMP
+  //    are either unset or one of them points to a valid place to create temporary
+  //    files and directories."
+  if (process.platform === 'win32') {
+    delete envCopy['TMP'];
+    delete envCopy['TMPDIR'];
+    delete envCopy['TEMP'];
+  }
+
```

Comment:
At least one key part of the change is missing here (again, it came in via multiple PRs, not just one of them). The `envCopy` set above must be passed to `spawnSync` in order for it to accomplish anything.

https://github.com/rstudio/rstudio/pull/12967